go/libkb/uid.go

```diff
@@ -48,10 +48,32 @@ func UIDArg(uid keybase1.UID) HTTPValue {
 	return S{Val: uid.String()}
 }
 
+// GetUIDByNormalizedUsername returns UID for normalized username. Works
+// offline for all usernames.
+func GetUIDByNormalizedUsername(g *GlobalContext, username NormalizedUsername) keybase1.UID {
+	uid := g.UIDMapper.MapHardcodedUsernameToUID(username)
+	if uid.Exists() {
+		return uid
+	}
+	return usernameToUIDPreserveCase(username.String())
+}
+
+// GetUIDByUsername returns UID for username strings with potentially
+// mixed letter casing. Works offline for all usernames.
+func GetUIDByUsername(g *GlobalContext, username string) keybase1.UID {
+	return GetUIDByNormalizedUsername(g, NewNormalizedUsername(username))
+}
+
+// NOTE: Use the high level API above instead of any of the following. The
+// hilvl API handles both UIDS for old, potentially incorrectly hashed
+// usernames, as well as new, correct UIDs.
+//
+// tldr: you probably want to use GetUID* functions, instead of UsernameToUID*.
+
 // UsernameToUID works for users created after "Fri Feb  6 19:33:08 EST 2015",
 // with some exceptions, since we didn't ToLower() for all UIDs
 func UsernameToUID(s string) keybase1.UID {
```

Comment:
Ideally we would unexport this one as well but it's used in a lot of tests right now